### DISCUSSION TITLE
LyraT-mini v1_2 testing.

### DISCUSSION THREAD

**Discussion by smacl61** [pschatzmann/arduino-audiokit#89] [Q&A] [answered]
Hello, I am in the process of testing the Lyrat-mini with es8311 codec. I'm just using the first input sketch. The program seems to run fine but the input data starts with 0 with the initial data block, then a burst of data up to +-70 then settles to what may be noise +-3 . The left/right channel data is the same being this is is a mono codec, I assume that is the expected result. I've used AUDIO_HAL_ADC_INPUT_LINE1 and 2 but the results are the same. 

I see that you had a similar issue with the es8388 where the mic gain was not perceivable and you have a fix for that. Possibly I have to select between the input mic modes, or apply a similar fix?

Any hints where I should start debugging this? 

Thanks,
Scott

**Comment by smacl61** [comment]
A few updates here. I did some tests to see what was working and here's what I found:
- I2C is working with the codec. 
- Mic is in analog mode. 
- Mic gain set manually in driver init doesn't change anything. 
- Tried output example and it produces the 1000 hz tone although choppy but confirms I2S.

**Comment by pschatzmann** [comment, marked as answer]
I am not sure how the mini is working exactly, since I don't own one myself

My recommendation would have been to set the gain by calling es8311_set_mic_gain() and use AUDIO_HAL_ADC_INPUT_LINE1. 
For testing try to record a sine tone, this helps to visually confirm that you get a proper signal.

I noticed that unlike the regular Lyrat there are 2 different i2s ports. It seems that the microphone is connected to the ES7243 and not to the ES8311.

Did you try the input on port 1 (setting i2s_num) instead of the default 0 port?

**Reply by smacl61** [reply]
Thanks for the fast response! Yes the second i2s is for the other adc but I hope to not use it and I'm not certain why they included it other than for better duplex communication.  But I didn't look into the setting i2s_num issue so I'll look into it. On my version the schematic the mic does go to both chips but I don't see anything in the audiokit that would select the es7243 by default anyway. It's a mystery at this point.

**Comment by pschatzmann** [comment]
Looking at this diagram: https://espressif-docs.readthedocs-hosted.com/projects/esp-adf/en/latest/design-guide/dev-boards/board-esp32-lyrat-mini-v1.2.html#id7
I dont see any connection from the mic to the es8311.
The selection is via the i2s port number...

**Reply by smacl61** [reply]
Thanks Phil, I think you are correct. I looked closely and saw the 2 caps C21/22 are removed from the pcb separating the mic from ES8311. It may be there were issues with performance trying to do everything on a single codec so they routed it to a ES7243 and used a second i2s stream. Or perhaps a single mic cannot be tied to 2 ES chips simultaneously. 
I may hack the board and add them back to see if the mic can be used at all with the ES8311 as for my app I don't want to used 2 codecs and I'm not sure if your Audiokit was intended to run 2 drivers at the same time - but correct me if I'm wrong.

**Comment by pschatzmann** [comment]
My gut feeling is that it should work: but since I don't have any LyraT-mini you are the only one who can check this.

**Reply by smacl61** [reply]
Ok great news! It now works and really well. I tested it with some tones and forms distinct patterns in the numbers coming out rather quickly on the monitor. The codec is really impressive. 
So I will be testing simultaneous send receive as my project is a communication system like a walkie talkie. 
Also I have to say i'm really impressed with assembling a development system based on your projects including audio-tools, audiokit, embedded them in espressif/arduino-esp32 and vscode as my editor in Debian.
